### PR TITLE
Improvements for DNSProvider objects

### PR DESCRIPTION
**Pull request by rfranzke**
**What this PR does / why we need it**:
The `DNSProvider` objects should not include the `api.` prefix but only the cluster's base domain. This is now corrected. Also, purpose labels are added (`dns.gardener.cloud/realms="<shoot-namespace>,"`).

**Release note**:

```improvement operator
Gardener does now annotate `DNSProvider` objects for the external cluster domain(s) with `dns.gardener.cloud/realms="<shoot-namespace>,"`.
```
```improvement operator
The `DNSProvider`s for a shoot do now only include the shoot's base domain (without `api.` prefix).
```

### CODE REVIEW

File: pkg/operation/shoot/shoot.go [hunk 1]
```diff
@@ -357,14 +363,14 @@ func ComputeTechnicalID(projectName string, shoot *gardenv1beta1.Shoot) string {
 // ConstructInternalClusterDomain constructs the domain pointing to the kube-apiserver of a Shoot cluster
```
Comment (timuthy) [inline pkg/operation/shoot/shoot.go]: ~// ConstructInternalClusterDomain constructs the domain pointing to the kube-apiserver of a Shoot cluster~

File: pkg/operation/shoot/shoot.go [hunk 2]
```diff
@@ -129,6 +129,12 @@ func (s *Shoot) GetIngressFQDN(subDomain string) string {
 	return fmt.Sprintf("%s.%s.%s", subDomain, common.IngressPrefix, *(s.Info.Spec.DNS.Domain))
 }
 
+// GetAPIServerDomain returns the fully qualified domain name of for the api-server for the Shoot cluster. The
+// end result is 'api.<domain>'.
+func (s *Shoot) GetAPIServerDomain(domain string) string {
```
Comment (timuthy) [inline pkg/operation/shoot/shoot.go]: Any reason why you use a pointer receiver `s *Shoot` here?

File: charts/seed-dns/provider/templates/dnsprovider.yaml [hunk 3]
```diff
@@ -13,6 +13,10 @@ kind: DNSProvider
 metadata:
   name: {{ .Values.name }}
   namespace: {{ .Release.Namespace }}
+{{- if ne .Values.name "internal" }}
```
Comment (timuthy) [inline charts/seed-dns/provider/templates/dnsprovider.yaml]:
```suggestion
{{- if ne .Values.purpose "internal" }}
```

Did you mean that?

**Review by timuthy — CHANGES_REQUESTED** [review]

File: pkg/operation/shoot/shoot.go (hunk 2 above)
Comment (rfranzke) [reply pkg/operation/shoot/shoot.go]: Thanks, good point. Moved to `common` package.

File: charts/seed-dns/provider/templates/dnsprovider.yaml (hunk 3 above)
Comment (rfranzke) [reply charts/seed-dns/provider/templates/dnsprovider.yaml]: Nice catch, thank you!

**Review by timuthy — APPROVED** [review]
/lgtm

**Pull request merged**